pynpoint/processing/psfsubtraction.py

```diff
@@ -43,44 +45,76 @@ def __init__(self,
-                 pca_numbers: Union[range, List[int], np.ndarray] = range(1, 21),
+                 pca_numbers: Union[range,
+                                    List[int],
+                                    np.ndarray,
+                                    Tuple[range, range],
+                                    Tuple[List[int], List[int]],
+                                    Tuple[np.ndarray, np.ndarray]] = range(1, 21),
                  extra_rot: float = 0.,
-                 subtract_mean: bool = True) -> None:
+                 subtract_mean: bool = True,
+                 processing_type: str = 'ADI') -> None:
         """
         Parameters
         ----------
         name_in : str
-            Unique name of the module instance.
+            Name tag of the pipeline module.
         images_in_tag : str
-            Tag of the database entry with the science images that are read as input
+            Database entry with the images from which the PSF model will be subtracted.
         reference_in_tag : str
-            Tag of the database entry with the reference images that are read as input.
+            Database entry with the reference images from which the PSF model is created. Usually
+            ``reference_in_tag`` is the same as ``images_in_tag``, but a different dataset can be
+            used as reference images in case of RDI.
         res_mean_tag : str, None
-            Tag of the database entry with the mean collapsed residuals. Not calculated if set to
-            None.
+            Database entry where the the mean-collapsed residuals will be stored. The residuals are
+            not calculated and stored if set to None.
         res_median_tag : str, None
-            Tag of the database entry with the median collapsed residuals. Not calculated if set
-            to None.
+            Database entry where the the median-collapsed residuals will be stored. The residuals
+            are not calculated and stored if set to None.
         res_weighted_tag : str, None
-            Tag of the database entry with the noise-weighted residuals (see Bottom et al. 2017).
-            Not calculated if set to None.
+            Database entry where the the noise-weighted residuals will be stored (see Bottom et al.
+            2017). The residuals are not calculated and stored if set to None.
         res_rot_mean_clip_tag : str, None
             Tag of the database entry of the clipped mean residuals. Not calculated if set to
             None.
         res_arr_out_tag : str, None
-            Tag of the database entry with the derotated image residuals from the PSF subtraction.
-            The tag name of `res_arr_out_tag` is appended with the number of principal components
-            that was used. Not calculated if set to None. Not supported with multiprocessing.
+            Database entry where the derotated, but not collapsed, residuals are stored. The number
```

Comment:
Explain how this works in case of IFS data.

Reply:
Added the comment. Please check if it is understandable (as soon as I uploaded the changes).